template-sample-app/template/src/components/layers/KpiLayer.js

```diff
@@ -31,11 +31,13 @@ function getFillColor(f) {
 export default function KpiLayer() {
   const { kpiLayer } = useSelector((state) => state.carto.layers);
   const source = useSelector((state) => selectSourceById(state, kpiLayer?.source));
+  const DEFAULT_PROPS = useCartoProps(source);
```

Comment:
What is exactly useCartoProps, I feel like the naming is not very clear

Reply:
Indeed `useCartoProps`, calls `useViewport` hook, so it needs to be a React Hook also.

The main propose of the hook is to take common layer logic out of the layer file. Actually viewport features management and filtering.

Proposals:
- `useCartoLayerProps`
- `useCartoFeaturesProps`
- Another name reflecting viewport and filtering logic ?¿

Reply:
Yeah, I would try to be explicit with the name, like `useCartoLayerWithFilters`, and I am not sure about the name "DEFAULT_PROPS". The caps suggests they are kind of 'static'. What exact props are we returning? We should probably find another name beyond 'default'

Reply:
We are returning the following props:
```javascript
{
  onViewportChange: debounce(onViewportChange, 500),
  getFilterValue: applyFilter({ filters: source?.filters, returnedType: 'number' }),
  filterRange: [1, 1],
  extensions: [new DataFilterExtension({ filterSize: 1 })],
  updateTriggers: {
    getFilterValue: source?.filters
  },
}

Removing caps could be a good approach, and rename it for something like `cartoFilterProps`

const cartoFilterProps = useCartoLayerFilters(source);

Another point:
Will be a case where the user needs to specify another trigger value like this (or event overwrite the props we are defining here):
```javascript
return new CartoSQLLayer({
   ...cartoFilterProps,
   ...otherProps,
   updateTriggers: {
    getFillColor: trigger
    ...cartoFilterProps.updateTriggers    
   }
´´´

So the docs should reflect it as well.

Reply:
cartoFilterProps sounds much better